services/consensusalgo/benchmarkconsensus/leader.go

```diff
@@ -93,25 +94,31 @@ func (s *service) leaderGenerateNewProposedBlockUnderMutex() (*protocol.BlockPai
 
 	// get tx
 	txOutput, err := s.consensusContext.RequestNewTransactionsBlock(&services.RequestNewTransactionsBlockInput{
-		BlockHeight: s.lastCommittedBlockHeight() + 1,
+		BlockHeight:   s.lastCommittedBlockHeight() + 1,
+		PrevBlockHash: crypto.CalcTransactionsBlockHash(s.lastCommittedBlock.TransactionsBlock),
 	})
 	if err != nil {
 		return nil, err
 	}
+	if txOutput == nil || txOutput.TransactionsBlock == nil {
+		panic("invalid response consensusContext.RequestNewTransactionsBlock: missing TransactionsBlock")
```

Comment:
These are dev only. They can only be nil if our internal code forgets to pass these arguments. We should remove these panics before going to production, right now they help with debugging because the code is missing in many places